src/meander/syntax/delta.cljc

```diff
@@ -514,10 +512,7 @@
                         {:pattern xs
                          :meta (meta xs)}))
         (parse
-         `(~'pred coll?
-           ;; Will cause compiler to emit a useless vector?
-           ;; check.
-           (~'app vec [~@'(_ ...) ~@(rest xs) ~@'(. _ ...)]))
+         `(~'app vec (~'seqable ~@'(_ ...) ~@(rest xs) ~@'(. _ ...)))
```

Comment:
We need to keep the `coll?` check around otherwise `app vec` will `throw`.

Reply:
Good catch. I will make these changes soon. But once I land some type inference stuff, I think all of this will be unnecessary. Should land this week. Will wait to close this PR until then.